src/TSRRobot.cpp

```diff
@@ -0,0 +1,258 @@
+#include <or_ompl/TSRRobot.h>
+#include <or_ompl/or_conversions.h>
+#include <boost/make_shared.hpp>
+
+using namespace or_ompl;
+
+TSRRobot::TSRRobot(const std::vector<TSR::Ptr> &tsrs, const OpenRAVE::EnvironmentBasePtr &penv)
+    : _tsrs(tsrs), _penv(penv), _initialized(false) {
+
+ 
+}
+
+bool TSRRobot::construct() {
+
+    if(_initialized){
+        RAVELOG_DEBUG("[TSRRobot] Already initialized. Skipping construct.");
+        return _initialized;
+    }
+
+    _initialized = false;
+
+    // Create an emtpy robot of the correct type
+    _probot = RaveCreateRobot(_penv, "GenericRobot");
+    if( _probot.get() == NULL ){
+        RAVELOG_INFO("[TSRRobot] Failed to create robot of type GenericRobot");
+        return _initialized;
+    }
+
+    // TODO: mimic body
+
+    // Build the robot
+    std::vector<OpenRAVE::KinBody::LinkInfoConstPtr> link_infos;
+    std::vector<OpenRAVE::KinBody::JointInfoConstPtr> joint_infos;
+    std::vector<OpenRAVE::RobotBase::ManipulatorInfoConstPtr> manip_infos;
+    std::vector<OpenRAVE::RobotBase::AttachedSensorInfoConstPtr> sensor_infos;
+
+    const std::string bodyprefix = "Body";
+    int bodynumber = 1;
+    Eigen::Affine3d Tw0_e = Eigen::Affine3d::Identity();
+       
+    for(unsigned int i=0; i < _tsrs.size(); i++){
+
+        TSR::Ptr tsr = _tsrs[i];
+        Eigen::Matrix<double, 6, 2> Bw = tsr->getBounds();
+        
+        for(int j=0; j < 6; j++){
+            
+            // Don't add a body if there is no freedom in this dimension
+            if(Bw(j,0) == 0.0 && Bw(j,1) == 0.0){
+                continue;
+            }
+
+            // If the bounds are equal and non-zero, we should do something reasonable
+            //  For now, this isn't supported
+            if(Bw(j,0) == Bw(j,1)){
+                RAVELOG_FATAL("[TSRRobot] ERROR: TSR Chains are currently unable to deal with cases where two bounds are equal but non-zero, cannot robotize.\n");
+                return _initialized;
+            }
+
+            // Check for axis flip, marked by Bw values being backwards
+            bool bFlipAxis = false;
+            if(Bw(j,0) > Bw(j,1)){
+                Bw(j,0) = -Bw(j,0);
+                Bw(j,1) = -Bw(j,1);
+
+                bFlipAxis = true;
+            }
```

Comment:
I can't think of a legitimate reason to use the former. Can we remove this? Or print a warning if we hit this case?